src/utils/label.ts

```diff
@@ -87,20 +88,54 @@ function hasAncestorCustomElements(label: HTMLCalciteLabelElement, componentEl:
 export function connectLabel(component: LabelableComponent): void {
   const labelEl = findLabelForComponent(component.el);
 
-  if (!labelEl || onLabelClickMap.has(labelEl)) {
+  const addClickEventListenerToComponentLabel = () => {
+    component.labelEl = labelEl;
+    const boundOnLabelClick = onLabelClick.bind(component);
+    onLabelClickMap.set(component.labelEl, boundOnLabelClick);
+    component.labelEl.addEventListener(labelClickEvent, boundOnLabelClick);
+  };
+
+  if (onLabelClickMap.has(labelEl)) {
     return;
   }
 
-  component.labelEl = labelEl;
-  const boundOnLabelClick = onLabelClick.bind(component);
-  onLabelClickMap.set(component.labelEl, boundOnLabelClick);
-  component.labelEl.addEventListener(labelClickEvent, boundOnLabelClick);
+  if (labelEl) {
+    addClickEventListenerToComponentLabel();
+
+    //when label is added for the existing component, remove label event listener for this component
+    const componentLabelAddedEventListener = labelListenerMap.get(component);
+    if (componentLabelAddedEventListener) {
+      document.removeEventListener("labelAdded", componentLabelAddedEventListener);
+      labelListenerMap.delete(component);
+    }
+
+    console.log("componentLabelAddedEventListener", componentLabelAddedEventListener);
+  }
+
+  if (!labelEl) {
+    const componentLabelAddedEventListener = labelListenerMap.get(component);
+    if (!componentLabelAddedEventListener) {
+      const labelAddedListener = () => {
+        connectLabel(component);
+      };
+      document.addEventListener("labelAdded", labelAddedListener);
+      labelListenerMap.set(component, labelAddedListener);
```

Comment:
I don't think we need to store the listener because we know what the listener function will be.